core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1713,16 +1747,31 @@ case class LeaderIsrAndControllerEpoch(leaderAndIsr: LeaderAndIsr, controllerEpo
   }
 }
 
-object ControllerStats extends KafkaMetricsGroup {
-
-  private val _uncleanLeaderElectionRate = newMeter("UncleanLeaderElectionsPerSec", "elections", TimeUnit.SECONDS)
-  private val _leaderElectionTimer = new KafkaTimer(newTimer("LeaderElectionRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
-
-  // KafkaServer needs to initialize controller metrics during startup. We perform initialization
-  // through method calls to avoid Scala compiler warnings.
-  def uncleanLeaderElectionRate: Meter = _uncleanLeaderElectionRate
+private[controller] class ControllerStats extends KafkaMetricsGroup {
+  val uncleanLeaderElectionRate = newMeter("UncleanLeaderElectionsPerSec", "elections", TimeUnit.SECONDS)
+  val leaderElectionTimer = new KafkaTimer(newTimer("LeaderElectionRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val startTimer = new KafkaTimer(newTimer("ControllerStartRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val resignTimer = new KafkaTimer(newTimer("ControllerResignRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val topicCreationTimer = new KafkaTimer(newTimer("TopicCreationRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val topicDeletionTimer = new KafkaTimer(newTimer("TopicDeletionRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val partitionReassignmentTimer = new KafkaTimer(newTimer("PartitionReassignmentRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val autoLeaderBalancingTimer = new KafkaTimer(newTimer("AutoLeaderBalancingRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+  val manualLeaderBalancingTimer = new KafkaTimer(newTimer("ManualLeaderBalancingRateAndTimeMs", TimeUnit.MILLISECONDS, TimeUnit.SECONDS))
+}
 
-  def leaderElectionTimer: KafkaTimer = _leaderElectionTimer
+sealed abstract class ControllerState(val value: Byte)
+
+object ControllerState {
+  case object Idle extends ControllerState(0)
+  case object Starting extends ControllerState(1)
+  case object Resigning extends ControllerState(2)
+  case object ElectingLeader extends ControllerState(3)
```

Comment:
The KIP says `BrokerChange` indeed. However, we have `LeaderElectionRateAndTimeMs` that seems to measure the same code path. So, I thought it would make sense to use a similar name.